common.mk

```diff
@@ -73,15 +73,15 @@ linux-common: linux-defconfig
 linux-defconfig-clean-common:
 	@rm -f $(LINUX_PATH)/.config
 
-LINUX_CLEAN_COMMON_FLAGS ?= $(LINUX_COMMON_FLAGS)
+#LINUX_CLEAN_COMMON_FLAGS ?= UNCOMMENT_AND_DEFINE_AS_NECESSARY
```

Comment:
This would be sufficient, the other stuff could be kept as it is in qemu.mk, hikey.mk etc... since we already have the correct ARCH flag there.

```
diff --git a/common.mk b/common.mk
index ae98c70..ece5ff5 100644
--- a/common.mk
+++ b/common.mk
@@ -73,7 +73,7 @@ linux-common: linux-defconfig
 linux-defconfig-clean-common:
        @rm -f $(LINUX_PATH)/.config
 
-LINUX_CLEAN_COMMON_FLAGS ?= $(LINUX_COMMON_FLAGS)
+LINUX_CLEAN_COMMON_FLAGS ?= CROSS_COMPILE=$(CROSS_COMPILE_NS_KERNEL)
        
 linux-clean-common: linux-defconfig-clean
        $(MAKE) -C $(LINUX_PATH) $(LINUX_CLEAN_COMMON_FLAGS) clean
```

qemu gives ...
```
make -C /home/jbech/devel/optee/build/../linux CROSS_COMPILE="/usr/bin/ccache /home/jbech/devel/optee/build/../toolchains/aarch32/bin/arm-linux-gnueabihf-" ARCH=arm clean
```

hikey gives ...
```
make -C /home/jbech/devel/optee/build/../linux CROSS_COMPILE="/usr/bin/ccache /home/jbech/devel/optee/build/../toolchains/aarch64/bin/aarch64-linux-gnu-" ARCH=arm64 clean
```

etc ...